Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -477,6 +486,22 @@ private extension ProductStore {
             }
         }
     }
+
+    func generateProductDescription(siteID: Int64,
+                                    name: String,
+                                    features: String,
+                                    languageCode: String,
+                                    completion: @escaping (Result<String, Error>) -> Void) {
+        let prompt = "Perform in-depth keyword research for a product on an e-commerce store and return a product description " +
+        "that includes as many keywords as possible and at least 200 words. " +
+        "The tone should be professional and without many questions. " +
+        "Try to make shorter sentences, using less difficult words to improve readability. " +
+        "Product name: \(name)\nFeatures: \(features)\nLanguage: \(languageCode)"
+        Task {
+            let result = await Result { try await generativeContentRemote.generateText(siteID: siteID, base: prompt) }
+            completion(result)
```

Comment:
❓ Should we make sure to call the completion block from the main thread? 

I am asking because,
1. This current detached `Task` isn't asking to be run in `MainActor`.
1. I think all our completion blocks from any `Store` is run on the main thread.

Reply:
From my understanding, it's up to the caller to decide whether it's necessary for the completion block to be called on the main thread using `@MainActor`. `Store` only enforces dispatching the action on the main thread. I don't know about the current state of Swift concurrency, in the past marking the function executed from a task call as `@MainActor` is the only way to be called on the main thread.

In https://github.com/woocommerce/woocommerce-ios/pull/9546/commits/f6c9ac1c30edc3ef38994550cbe8266dd72e7a3a I updated to call the completion block in `MainActor.run` as a precaution.